Index: or1200/rtl/verilog/or1200_genpc.v
===================================================================
--- or1200.orig/rtl/verilog/or1200_genpc.v
+++ or1200/rtl/verilog/or1200_genpc.v
@@ -252,6 +252,10 @@ reg				genpc_refetch_r;
 	endcase
      end
 
+   // select async. value for pcreg after reset - PC jumps to the address selected
+   // after boot.
+   wire [31:0] pcreg_boot = boot_adr;
+
    //
    // PC register
    //
@@ -276,10 +280,6 @@ reg				genpc_refetch_r;
 	pcreg_default <=  pc[31:2];
      end
 
-   // select async. value for pcreg after reset - PC jumps to the address selected
-   // after boot.
-   wire [31:0] pcreg_boot = boot_adr;
-
    always @(pcreg_boot or pcreg_default or pcreg_select)
      if (pcreg_select)
        // async. value is selected due to reset state 
