Index: uart16550/rtl/verilog/uart_regs.v
===================================================================
--- uart16550.orig/rtl/verilog/uart_regs.v
+++ uart16550/rtl/verilog/uart_regs.v
@@ -363,6 +363,8 @@ reg 										tf_push;
 reg 										rf_pop;
 wire [`UART_FIFO_REC_WIDTH-1:0] 	rf_data_out;
 wire 										rf_error_bit; // an error (parity or framing) is inside the fifo
+wire 					rf_overrun;
+wire 					rf_push_pulse;
 wire [`UART_FIFO_COUNTER_W-1:0] 	rf_count;
 wire [`UART_FIFO_COUNTER_W-1:0] 	tf_count;
 wire [2:0] 								tstate;
