Index: uart16550/rtl/verilog/uart_regs.v
===================================================================
--- uart16550.orig/rtl/verilog/uart_regs.v
+++ uart16550/rtl/verilog/uart_regs.v
@@ -486,13 +486,22 @@ always @(posedge clk or posedge wb_rst_i
 	if (wb_rst_i)
 	begin
 		ier <= 4'b0000; // no interrupts after reset
+`ifdef PRESCALER_PRESET_HARD
+		dl[`UART_DL2] <= `PRESCALER_HIGH_PRESET;
+`else
 		dl[`UART_DL2] <= 8'b0;
+`endif
 	end
 	else
 	if (wb_we_i && wb_addr_i==`UART_REG_IE)
 		if (dlab)
 		begin
-			dl[`UART_DL2] <= wb_dat_i;
+			dl[`UART_DL2] <=
+`ifdef PRESCALER_PRESET_HARD
+			dl[`UART_DL2];
+`else
+		   wb_dat_i;
+`endif
 		end
 		else
 			ier <= wb_dat_i[3:0]; // ier uses only 4 lsb
@@ -535,7 +544,11 @@ always @(posedge clk or posedge wb_rst_i
 always @(posedge clk or posedge wb_rst_i)
 	if (wb_rst_i)
 	begin
+`ifdef PRESCALER_PRESET_HARD
+		dl[`UART_DL1]  <= `PRESCALER_LOW_PRESET;
+`else
 		dl[`UART_DL1]  <= 8'b0;
+`endif
 		tf_push   <= 1'b0;
 		start_dlc <= 1'b0;
 	end
@@ -543,7 +556,11 @@ always @(posedge clk or posedge wb_rst_i
 	if (wb_we_i && wb_addr_i==`UART_REG_TR)
 		if (dlab)
 		begin
+`ifdef PRESCALER_PRESET_HARD
+			dl[`UART_DL1] <= dl[`UART_DL1];
+`else
 			dl[`UART_DL1] <= wb_dat_i;
+`endif
 			start_dlc <= 1'b1; // enable DL counter
 			tf_push <= 1'b0;
 		end
